kinesis/src/main/scala/akka/stream/alpakka/kinesis/CommittableRecord.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.kinesis
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.stream.alpakka.kinesis.impl.ShardProcessor
+import software.amazon.kinesis.lifecycle.ShutdownReason
+import software.amazon.kinesis.retrieval.KinesisClientRecord
+import software.amazon.kinesis.retrieval.kpl.ExtendedSequenceNumber
+
+import scala.util.Try
+
+class CommittableRecord @InternalApi private[kinesis] (
+    val shardId: String,
+    val recordProcessorStartingSequenceNumber: ExtendedSequenceNumber,
+    val millisBehindLatest: Long,
+    val record: KinesisClientRecord,
+    recordProcessor: ShardProcessor,
+    checkpointer: KinesisClientRecord => Unit
+) {
+
+  val sequenceNumber: String = record.sequenceNumber()
+  val subSequenceNumber: Long = record.subSequenceNumber()
+
+  def recordProcessorShutdownReason: Option[ShutdownReason] =
+    recordProcessor.shutdown
+
+  def canBeCheckpointed: Boolean =
+    !recordProcessorShutdownReason.contains(ShutdownReason.LEASE_LOST)
+
+  def tryToCheckpoint(): Try[Done] =
+    Try(checkpointer(record)).map(_ => Done)
```

Comment:
Are these internal API?

Reply:
It wraps a call to the internal checkpoint API, yes.

https://github.com/akka/alpakka/pull/1667/files#diff-877d82ee076716a8e876b7a6551328d0R47-R49